### PR TITLE
Set up `metalsmith.watch()` with partial build support

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR adds partial Metalsmith rebuilds as shown in https://github.com/alphagov/govuk-design-system/pull/2750#issuecomment-1600843431

For example, CSS changes are injected without a browser reload

Other improvements that came from https://github.com/alphagov/govuk-design-system/pull/2934 have been split out into:

* https://github.com/alphagov/govuk-design-system/pull/2942

### CODE REVIEW

**Comment by colinrotherham** [comment]
This is likely blocked due to Metalsmith `.watch()` issues with async plugins

The [`build()`](https://metalsmith.io/api/#Metalsmith+build) callback never fires with both [`watch()`](https://metalsmith.io/api/#Metalsmith+watch) configured and async plugins like `rollup()` present

>### `metalsmith.watch([options])`
>https://metalsmith.io/api/#Metalsmith+watch

**Comment by colinrotherham** [comment]
@domoscargin [Metalsmith v2.6.1](https://github.com/metalsmith/metalsmith/releases/tag/v2.6.1) was released yesterday 🙌

They’ve fixed a [watch mode bug we raised](https://github.com/metalsmith/metalsmith/issues/389)

So this PR is now ready for review 😊

**Review by domoscargin — APPROVED** [review]
This is awesome! Everything feels a lot quicker.

My comment is unblocking. Only difference in the output is the addition of `<meta name="robots" content="noindex, nofollow">` on all pages, and the removal of the file hashes.

File: tasks/start.js [hunk 1]
```diff
@@ -1,22 +1,52 @@
...
-metalsmith.build(function (err, files) {
-  if (err) { throw err }
-})
+
+let bs
+
+metalsmith
+
+  // Configure sources to watch for rebuilds
+  .watch([
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply tasks/start.js:14]: @domoscargin Yeah, sadly I can't hot reload the Node.js process (running Metalsmith) that easily 😣

[the remark this answers is not in the document]
Comment (domoscargin) [reply tasks/start.js:14]: What, no `metalsmith-nodemon` plugin? 😂

[the remark this answers is not in the document]
Comment (colinrotherham) [reply tasks/start.js:14]: PRs welcome 😉

**Comment by colinrotherham** [comment]
> My comment is unblocking. Only difference in the output is the addition of `<meta name="robots" content="noindex, nofollow">` on all pages, and the removal of the file hashes.

Thanks @domoscargin 🙌

I'm not going to merge yet as `production` builds don't get `<meta name="robots">`

That sounds great, except Netlify also uses a `production` build (where we _do_ want the meta tag)

**Comment by colinrotherham** [comment]
I've added the [**Netlify** build `CONTEXT` environment variable](https://docs.netlify.com/configure-builds/environment-variables/#build-metadata) in https://github.com/alphagov/govuk-design-system/pull/2865/commits/4d62eba36fdc3d484c45c1d2a544bcb89269389a

1. Environment `CONTEXT` – Netlify deployment context
2. Environment `NODE_ENV` – Node.js build environment

I've kept both as we _could_ debug a development build in production one day

**Pull request merged**